elastic/kibana
# Make Watcher use Query watches API instead of .watches index

Fixes https://github.com/elastic/kibana/issues/152142

## Summary

This PR makes Watcher fetch watches using the [Query watches API](https://www.elastic.co/guide/en/elasticsearch/reference/8.6/watcher-api-query-watches.html) instead of sending a search request to the `.watches` index which is not recommended anymore.

**How to test**

1. Go to Stack Management -> Watcher.
2. Create a couple of watches (maybe a few threshold alerts and some advanced watches as well).
3. Go to the list of all watches and verify that it works as expected and all watches are correctly displayed.
